internal/app/machined/pkg/system/services/bootkube.go

```diff
@@ -0,0 +1,210 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package services
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"net"
+	"net/url"
+	"os"
+
+	"github.com/kubernetes-incubator/bootkube/pkg/asset"
+	"github.com/kubernetes-incubator/bootkube/pkg/tlsutil"
+	"github.com/pkg/errors"
+
+	"github.com/talos-systems/talos/internal/app/machined/internal/bootkube"
+	"github.com/talos-systems/talos/internal/app/machined/pkg/system/conditions"
+	"github.com/talos-systems/talos/internal/app/machined/pkg/system/runner"
+	"github.com/talos-systems/talos/internal/app/machined/pkg/system/runner/goroutine"
+	"github.com/talos-systems/talos/pkg/config"
+	"github.com/talos-systems/talos/pkg/constants"
+)
+
+// Bootkube implements the Service interface. It serves as the concrete type with
+// the required methods.
+type Bootkube struct{}
+
+// ID implements the Service interface.
+func (b *Bootkube) ID(config config.Configurator) string {
+	return "bootkube"
+}
+
+// PreFunc implements the Service interface.
+func (b *Bootkube) PreFunc(ctx context.Context, config config.Configurator) (err error) {
+	return generateAssets(config)
+}
+
+// PostFunc implements the Service interface.
+func (b *Bootkube) PostFunc(config config.Configurator) error {
+	return nil
+}
+
+// DependsOn implements the Service interface.
+func (b *Bootkube) DependsOn(config config.Configurator) []string {
+	deps := []string{"etcd"}
+
+	return deps
+}
+
+// Condition implements the Service interface.
+func (b *Bootkube) Condition(config config.Configurator) conditions.Condition {
+	return nil
+}
+
+// Runner implements the Service interface.
+func (b *Bootkube) Runner(config config.Configurator) (runner.Runner, error) {
+	return goroutine.NewRunner(config, "bootkube", bootkube.NewService().Main), nil
+}
+
+// nolint: gocyclo
+func generateAssets(config config.Configurator) (err error) {
+	if err = os.MkdirAll("/etc/kubernetes/manifests", 0644); err != nil {
```

Comment:
This strikes me as either an odd placement or a mismatch.  Nothing else here seems to use this directory, as declared.  There _is_ a write to `constants.AssetsDirectory`, however.

Reply:
Agree. Can fix in follow up.